src/Conversions.cc

```diff
@@ -341,6 +359,28 @@ sdf::Geometry gz::sim::convert(const msgs::Geometry &_in)
     meshShape.SetSubmesh(_in.mesh().submesh());
     meshShape.SetCenterSubmesh(_in.mesh().center_submesh());
 
+    sdf::ConvexDecomposition convexDecomp;
+    bool setConvexDecomp = false;
+    for (int i = 0; i < _in.header().data_size(); ++i)
+    {
+      auto data = _in.header().data(i);
+      if (data.key() == "optimization" && data.value_size() > 0)
+      {
+        meshShape.SetOptimization(data.value(0));
+      }
+      if (data.key() == "max_convex_hulls" && data.value_size() > 0)
+      {
+        convexDecomp.SetMaxConvexHulls(std::stoul(data.value(0)));
+        setConvexDecomp = true;
+      }
+      if (data.key() == "voxel_resolution" && data.value_size() > 0)
+      {
+        convexDecomp.SetVoxelResolution(std::stoul(data.value(0)));
+        setConvexDecomp = true;
+      }
+    }
+    if (setConvexDecomp)
+      meshShape.SetConvexDecomposition(convexDecomp);
```

Comment:
Nit: Could you remove the `if` and let just ` meshShape.SetConvexDecomposition(convexDecomp);`?

Reply:
yep removed. [c74e7c1](https://github.com/gazebosim/gz-sim/pull/2352/commits/c74e7c1a6ff7f9e6c8d1c437bdb1bdc3e6c9addf)